compiler/include/ResolutionCandidate.h

```diff
@@ -30,6 +30,48 @@ class CallInfo;
 class FnSymbol;
 class Symbol;
 
+typedef enum {
+  // These are in order of severity, for failedCandidateIsBetterMatch.
+
+  RESOLUTION_CANDIDATE_MATCH,
+
+  // Types do not match but are related (e.g. borrowed C vs unmanaged C)
+  RESOLUTION_CANDIDATE_TYPE_RELATED,
+
+  // Types do not match but are in the same general category
+  // (class-ish things, numeric things, ...)
+  RESOLUTION_CANDIDATE_TYPE_SAME_CATEGORY,
+
+  // Where clause does not match
+  RESOLUTION_CANDIDATE_WHERE_FAILED,
+
+  // Types do not match and are in different categories
+  RESOLUTION_CANDIDATE_UNRELATED_TYPE,
+
+  // Formal is param but actual is not
+  RESOLUTION_CANDIDATE_NOT_PARAM,
+
+  // Formal is type but actual is not or vice versa
+  RESOLUTION_CANDIDATE_NOT_TYPE,
+
+
```

Comment:
The reason for having the second newline above is unclear. Maybe leave just 1 empty line?